b2/bucket.py

```diff
@@ -99,19 +99,38 @@ class Bucket(object):
     MAX_UPLOAD_ATTEMPTS = 5
     MAX_LARGE_FILE_SIZE = 10 * 1000 * 1000 * 1000 * 1000  # 10 TB
 
-    def __init__(self, api, id_, name=None, type_=None):
+    def __init__(self, api, id_, name=None, type_=None, bucket_info=None, revision=None):
         self.api = api
         self.id_ = id_
         self.name = name
         self.type_ = type_
+        self.bucket_info = bucket_info or {}
+        self.revision = revision
 
     def get_id(self):
         return self.id_
 
+    def set_info(self, new_bucket_info, ifRevisionIs=None):
+        account_id = self.api.account_info.get_account_id()
+        return self.api.session.update_bucket(
+            account_id, self.id_, bucket_info=new_bucket_info, ifRevisionIs=ifRevisionIs
+        )
+
     def set_type(self, type_):
         account_id = self.api.account_info.get_account_id()
         return self.api.session.update_bucket(account_id, self.id_, type_)
 
+    def update(self, bucket_type=None, bucket_info=None, lifecycle_rules=None, if_revision_is=None):
```

Comment:
I suggest that `set_type` and `set_info` use `update` internally, rather than directly interacting with `self.api.session.update_bucket`. It prevents `account_id = self.api.account_info.get_account_id()` from being copied all over the place, which may not seem like much now, but later when any changes are done to `update_bucket()`, it'd be nice to only need to change one area in `Bucket`.